AirCasting/CoreData/NSManagedObjectContext+Utils.swift

```diff
@@ -39,6 +39,22 @@ extension NSManagedObjectContext {
         return new
     }
     
+    // Checks if stream exists, if not, creates a new one
+    func newOrExisting<T: NSManagedObject>(streamID: MeasurementStreamID) throws -> T  {
+        let className = NSStringFromClass(T.classForCoder())
+        let fetchRequest = NSFetchRequest<T>(entityName: className)
+        fetchRequest.predicate = NSPredicate(format: "id == \(streamID)")
+        
+        let results = try self.fetch(fetchRequest)
+        if let existing  = results.first {
+            return existing
+        }
+        
+        let new: T = T(context: self)
+        new.setValue(streamID, forKey: "id")
```

Comment:
I think we can use `#keyPath` in all places in this class instead of a plain string because we're dealing with `NSManagedObject` subclasses and those have `@objc` properties. Advantage of that approach is that we're getting compiler check for free and if we ever change for example `sensorName` to `sensorLabel` we'd get an error here. And now if we do this the code would compile and we'd probably get to know about the problem by crashes reported in the production app.

Reply:
I tried writing `#keypath(MeasurementStreamEntity.id)` but it won't compile because `id` is defined as a Swift property. I've tried adding `@objc` to it but it's type - `MeasurementStreamID` can't be represented in Obj-C (because it's a Swift typealias and from what I read, they can't be represented in Objc).
Not sure then if there's any better way to do this 🤷🏻‍♀️.